### PR TITLE
react-virtualized: Update types for WindowScroller

### PR DESCRIPTION
**Pull request by thevtm**
[WindowScroller docs](https://github.com/bvaughn/react-virtualized/blob/master/docs/WindowScroller.md)

### CODE REVIEW

**Comment by thevtm** [comment]
The issue is in the `react-virtualized-select` package.

### Error
```
.../DefinitelyTyped/types/react-virtualized-select/index.d.ts
ERROR: 25:18  expect  TypeScript@next compile error: 
Interface 'VirtualizedSelectProps' incorrectly extends interface 'ReactSelectProps'.
  Types of property 'optionRenderer' are incompatible.
    Type '((options: VirtualizedOptionRenderOptions<any>) => Element) | undefined' is not assignable to type 'OptionRendererHandler | undefined'.
      Type '(options: VirtualizedOptionRenderOptions<any>) => Element' is not assignable to type 'OptionRendererHandler | undefined'.
        Type '(options: VirtualizedOptionRenderOptions<any>) => Element' is not assignable to type 'OptionRendererHandler'.
          Types of parameters 'options' and 'option' are incompatible.
            Type 'Option<OptionValues>' is not assignable to type 'VirtualizedOptionRenderOptions<any>'.
              Property 'focusedOption' is missing in type 'Option<OptionValues>'.
```

:bug: Property `focusedOption` (from `VirtualizedOptionRenderOptions<T>`) is missing in type `Option<OptionValues>`.

Basically `VirtualizedOptionRenderOptions<T>` cannot be assigned to type `Option<OptionValues>`.

### Option\<OptionValues\>
```typescript
    interface Option<TValue = OptionValues> {
        /** Text for rendering */
        label?: string;
        /** Value for searching */
        value?: TValue;
        /**
         * Allow this option to be cleared
         * @default true
         */
        clearableValue?: boolean;
        /**
         * Do not allow this option to be selected
         * @default false
         */
        disabled?: boolean;
    }
```

### VirtualizedOptionRenderOptions\<T\>
```typescript
export interface VirtualizedOptionRenderOptions<T> {
    focusedOption: T;
    focusedOptionIndex: number;
    focusOption(option: T): void;
    key: string;
    labelKey: keyof T;
    listProps: ListProps;
    option: T;
    options: T[];
    selectValue(option: T): void;
    style: Record<string, string>;
    valueArray: T[];
}
```

### Solution
Make `VirtualizedOptionRenderOptions<T>` extend `Option<OptionValues>`.

`export interface VirtualizedOptionRenderOptions<T> extends Option<T> `

File: types/react-virtualized/dist/es/WindowScroller.d.ts [hunk 1]
```diff
@@ -28,23 +32,29 @@ export type WindowScrollerProps = {
...
-        scrollElement: HTMLElement
+        scrollElement: Validator<HTMLElement>,
+        scrollingResetTimeInterval: Validator<number>
     };
 
     static defaultProps: {
         onResize: () => {},
-        onScroll: () => {}
+        onScroll: () => {},
+        // scrollElement: window,
```
Comment (minestarks) [inline types/react-virtualized/dist/es/WindowScroller.d.ts]: Please remove comment or add back the code

**Comment by minestarks** [comment]
Author @seansfkelley any comment

**Comment by seansfkelley** [comment]
Compile errors are probably unrelated to this change. https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19194

**Comment by thevtm** [comment]
@minestarks done!

@seansfkelley so remove the `react-virtualized-select` commit?

**Review by minestarks — APPROVED** [review]

**Comment by minestarks** [comment]
Will merge if @seansfkelley approves

**Comment by seansfkelley** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19194 merged a couple days ago so presumably if you merge/rebase and remove the changes to react-virtualized-select it should still compile...?

**Comment by thevtm** [comment]
Rebase and remove the changes to react-virtualized-select.

**Pull request merged**